docs/setting-up-custom-ingress-gateway.md

```diff
@@ -0,0 +1,254 @@
+# Setting Up Custom Ingress Gateway
+
+Knative uses a shared Gateway to serve all incoming traffic within Knative
+service mesh, which is the "knative-shared-gateway" Gateway under
+"knative-serving" namespace. By default, we use Istio gateway service `istio-ingressgateway`
+under "istio-system" namespace as its underlying service. You can replace the
+service with that of your own as follows.
+
+## Step 1: Create Gateway Service and Deployment Instance
+
+You'll need to create the gateway service and deployment instance to handle traffic first.
+The simplest way should be making a copy of the Gateway service template in [Istio release](https://github.com/istio/istio/releases).
+
+Here is an example:
+
+```
```

Comment:
Can we instead provide some bash command instead?  Either way we will run into staleness issue, which may be fine for an example, but the substitution command will (hopefully) explain the idea better since it's not obvious from the yaml what the difference.

That's not blocking, we can follow up with another PR.

Reply:
I have a one-liner here 

https://github.com/knative/serving/pull/2546/files#diff-d11ce1cfee915c30534db0ffe418081f